applications/nrf_desktop/src/modules/usb_state_pm.c

```diff
@@ -58,7 +75,23 @@ static bool app_event_handler(const struct app_event_header *aeh)
 			power_manager_restrict(MODULE_IDX(MODULE), POWER_MANAGER_LEVEL_ALIVE);
 			break;
 		case USB_STATE_DISCONNECTED:
-			power_manager_restrict(MODULE_IDX(MODULE), POWER_MANAGER_LEVEL_MAX);
+			if (CONFIG_DESKTOP_USB_PM_RESTRICT_REMOVE_DELAY_MS > 0) {
+				/* Remove the restriction after the delay to allow the user to
+				 * take actions like e.g. restart BLE advertising without going
+				 * through system off.
+				 */
+				(void)k_work_schedule(
+					&pm_restrict_remove_work,
+					K_MSEC(CONFIG_DESKTOP_USB_PM_RESTRICT_REMOVE_DELAY_MS));
+			} else {
+				power_manager_restrict(MODULE_IDX(MODULE), POWER_MANAGER_LEVEL_MAX);
+			}
+			/* When disconnecting USB cable, the USB_STATE_SUSPENDED event may
+			 * or may not appear before the USB_STATE_DISCONNECTED event. Force power
+			 * down for application to behave consistently even if the
+			 * USB_STATE_SUSPENDED event doesn't appear.
+			 */
+			force_power_down();
```

Comment:
Consider also instantly updating restriction to `POWER_MANAGER_LEVEL_SUSPENDED` if we delay removing CAF power manager restrictions (to make it fully consistent)